### PR TITLE
Add skipmissing() to skip missing values

### PR DESCRIPTION
**Pull request by nalimilan**
Currently defined in Missings.jl. See design discussion at https://github.com/JuliaData/Missings.jl/issues/53.

We should be able to get rid of the specialized `AbstractArray` methods once the compiler handles `Union`s more efficiently, but benchmarking shows that it currently makes a dramatic difference.

### CODE REVIEW

File: doc/src/manual/missing.md [hunk 1]
```diff
@@ -296,6 +296,46 @@ since type constructors fall back to convert methods.
...
+julia> sum([1, missing])
+missing
+
+```
+
+In this situation, use the [`skipmissing`](@ref) function to skip missing values:
+```doctest
+julia> sum(skipmissing([1, missing]))
+1
+
```
Comment (kshyatt) [inline doc/src/manual/missing.md]: Extra line?

Comment (kshyatt) [reply doc/src/manual/missing.md]: Also shouldn't these be ```jldoctest and not ```doctest?

Comment (nalimilan) [reply doc/src/manual/missing.md]: Thanks for spotting this! Actually, I had made the same mistake in the whole file...

**Comment by nalimilan** [comment]
I'll merge tomorrow barring objections.

**Review by StefanKarpinski — APPROVED** [review]

**Pull request merged**